src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +333,21 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            if (colonIndex == -1)
+            {
+                throw new InvalidOperationException("TimeSpan.ToString() should contain at least one colon.");
```

Comment:
@davidsh These methods are internal and shouldn't throw under normal circumstances, though (as @stephentoub commented below). Would you recommend changing these to `Debug.Asserts`?